### PR TITLE
Add refund and redeem to swap

### PR DESCRIPTION
**Pull request by i3wgnit**
### Description

Implements `refundSwap` and `redeemSwap` methods in BitcoinSwapProvider and EthereumSwapProvider.

### Parts

- [x] Implement `redeemSwap`
- [x] Implement `refundSwap`
- [x] Implement Relevant tests

### Considerations

- Rename for `redeemSwap` to `generateRedeemSwap`
- Rename `refundSwap` to `generateRefundSwap`
- Rename `generateSwap` to `generateInitSwap`

### CODE REVIEW

File: src/providers/bitcoin/BitcoinSwapProvider.js [hunk 1]
```diff
@@ -2,36 +2,72 @@ import Provider from '../../Provider'
 import { addressToPubKeyHash } from './BitcoinUtil'
 
 export default class BitcoinSwapProvider extends Provider {
+  _padHexStart (hex, length) {
```
Comment (monokh) [inline src/providers/bitcoin/BitcoinSwapProvider.js]: can this be in a util somewhere instead?

**Comment by monokh** [comment]
@i3wgnit now that we are doing quite a bit of script and bytecode serialisation, i'm wondering, are there any small libraries that are well tested that can help us do this? Feel like this side of it is getting quite hard to read. Not a necessity but just to get us thinking

**Comment by i3wgnit** [comment]
Bitcoinjslib imports a few small libraries here and there. I am not too familiar with them, but I have gone to them and tried to reverse engineer a few processes.

File: src/providers/bitcoin/BitcoinSwapProvider.js (hunk 1 above)
Comment (i3wgnit) [reply src/providers/bitcoin/BitcoinSwapProvider.js]: We can put this in crypto?

Comment (harshjv) [reply src/providers/bitcoin/BitcoinSwapProvider.js]: IMO leave it here for now. we will rearrange it when we reuse it somewhere else.

**Review by harshjv — COMMENTED** [review]
Waiting for the tests

**Pull request merged**